src/components/Buttons/Button.js

```diff
@@ -0,0 +1,266 @@
+        ':active': {
+          opacity: '0.8'
+        }
+      }
+    },
+
+    secondary: {
+      base: {
+        backgroundColor: 'transparent',
+        color: action,
+        border: `1px solid ${action}`,
+
+        ':hover': {
+          color: actionHover,
+          border: `1px solid ${actionHover}`
+        },
+        ':active': {
+          color: actionHover,
+          border: `1px solid ${actionHover}`
+        }
+      },
+
+      inverted: {
+        color: primaryBackground,
+        border: `1px solid ${primaryBackground}`,
+
+        ':hover': {
+          opacity: '0.9'
+        },
+        ':active': {
+          opacity: '0.8'
+        }
+      }
+    },
+
+    coupon: {
+      base: {
+        backgroundColor: colors.WHITE,
+        color: colors.RED_500,
+        border: `1px dashed ${colors.GRAY_88}`,
+
+        ':hover': {
+          border: `1px dashed ${colors.RED_600}`,
+          color: colors.RED_600
+        },
+        ':active': {
+          border: `1px dashed ${colors.RED_700}`,
+          color: colors.RED_700
+        }
+      }
+    },
+
+    disabled: {
+      base: {
+        backgroundColor: colors.GRAY_74,
+        color: colors.WHITE,
+        cursor: 'not-allowed'
+      }
+    }
+  }
+}
+
+const Button = props => {
+  // Disabled styles override all other snacks-specific button styles
+  const snacksStyle = props.disabled ? 'disabled' : props.snacksStyle
+  const snacksStyles = getSnacksStyles(props)
+
+  const ElementType = props.href ? 'a' : 'button'
+
+  const finalProps = {
+    disabled: props.disabled,
+    tabIndex: props.tabIndex,
+    type: props.type,
+    style: [
+      baseStyles,
+      sizeStyles[props.size],
+      snacksStyles[snacksStyle].base,
+      props.inverted && snacksStyles[snacksStyle].inverted,
+      ElementType === 'a' && linkStyles,
+      props.styles
+    ],
+    onClick: e => {
+      e.preventDefault()
+      props.onClick()
+    },
+    ...props.extraProps
+  }
+  if (props.href) {
+    finalProps.href = props.href
+  }
+
+  let icon = props.icon
+  if (typeof props.icon === 'string') {
+    icon = <Icon name={props.icon} />
+  }
+
+  if (icon && props.iconPosition === 'left') {
+    return (
+      <ElementType {...finalProps}>
+        {icon}
+        <span style={{ ...spacing.MARGIN_RIGHT_XS }} />
+        {props.children}
+      </ElementType>
+    )
+  }
+
+  if (icon && props.iconPosition === 'right') {
+    return (
+      <ElementType {...finalProps}>
+        {props.children}
+        <span style={{ ...spacing.MARGIN_LEFT_XS }} />
+        {icon}
+      </ElementType>
+    )
+  }
+
+  return <ElementType {...finalProps}>{props.children}</ElementType>
+}
+
+Button.propTypes = {
+  /** Size of the button. */
+  size: PropTypes.oneOf(['tiny', 'small', 'standard', 'large']),
+
+  /** Snacks button variants. */
+  snacksStyle: PropTypes.oneOf(['primary', 'secondary', 'coupon']),
+
+  /** Optional style overrides. */
+  styles: PropTypes.object,
```

Comment:
I think we agreed to use `style` singular in Snacks to make it feel more like a regular style attribute.

Our other components use variations of both, so we'll need to update those some day :D

Reply:
I can get behind this